src/core/src/main/java/org/apache/jmeter/samplers/JMeterThreadBoundSampleListener.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.samplers;
+
+import org.apache.jmeter.threads.JMeterThread;
+
+/**
+ * Marker interface that informs {@link SampleListener} must be executed within {@link JMeterThread}
+ * @since 5.4
+ */
+public interface JMeterThreadBoundSampleListener extends SampleListener {
```

Comment:
Frankly speaking, I do not like `marker` interfaces. They provide very little way to configure the behavior, and they can't be "unimplemented". In other words, I can't create a subclass and adjust the marker.

Could we stop using marker interfaces and use something else instead?

Reply:
What do you propose as an alternative ?

Reply:
Annotations?

Reply:
Don't you think that in the future those interface could end up with some methods ? 
I mean JMeterThreadBoundSampleListener could contain some methods to access JMeterVariables, JMeterContext...

For now it is true that an Annotation would be better.

Reply:
>Don't you think that in the future those interface could end up with some methods ?

If you add a method to an interface, you break backward compatibility as all the clients need to implement the method.

I guess the next big thing is to reduce memory consumption caused by thread plan cloning for each thread.
Adding new interfaces with methods makes it harder.

Reply:
Default method of interface avoid breaking backward compat. Alternatively we can introduced base classes or if we think there will never be a need to enrich the interface go for Annotation